src/AutoRest.CSharp.V3/Generation/Writers/ModelWriter.cs

```diff
@@ -274,14 +274,28 @@ private void WriteChoiceSchema(CodeWriter writer, EnumType schema)
 
                     foreach (var choice in schema.Values)
                     {
-                        writer.Line($"private const {schema.BaseType} {choice.Declaration.Name}Value = {choice.Value.Value:L};");
+                        if (name == $"{choice.Declaration.Name}Value")
+                        {
+                            writer.Line($"private const {schema.BaseType} {choice.Declaration.Name}_Value = {choice.Value.Value:L};");
```

Comment:
Let's add a method instead. Something like `GetValueFieldName()` and put the logic there.

Reply:
I would also append an endex to the end (EnumValue1) I think it makes it a little bit more clear about what's happening than an underscore appearing in the middle.

Reply:
Added!